infinispan-client + micrometer combination doesn't compile in native

### Describe the bug

infinispan-client + micrometer combination doesn't compile in native when using Quarkus 3.0.3.Final / Quarkus main

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /Users/rsvoboda/.sdkman/candidates/java/22.3.1.r17-grl/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dlogging.initial-configurator.min-level=500 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dio.quarkus.caffeine.graalvm.recordStats=true -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=GB -J-Dfile.encoding=UTF-8 --features=io.quarkus.runner.Feature,io.quarkus.caffeine.runtime.graal.CacheConstructorsFeature,io.quarkus.runtime.graal.DisableLoggingFeature,io.quarkus.infinispan.client.runtime.graal.DisableLoggingFeature -J--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED -J--add-opens=java.base/java.text=ALL-UNNAMED -J--add-opens=java.base/java.io=ALL-UNNAMED -J--add-opens=java.base/java.lang.invoke=ALL-UNNAMED -J--add-opens=java.base/java.util=ALL-UNNAMED -H:+CollectImageBuildStatistics -H:ImageBuildStatisticsFile=infinispan-client-micrometer-1.0.0-SNAPSHOT-runner-timing-stats.json -H:BuildOutputJSONFile=infinispan-client-micrometer-1.0.0-SNAPSHOT-runner-build-output-stats.json -H:+AllowFoldMethods -J-Djava.awt.headless=true --no-fallback --link-at-build-time -H:+ReportExceptionStackTraces -H:-AddAllCharsets --enable-url-protocols=http,https -H:-UseServiceLoaderFeature -H:+StackTrace -H:AdditionalSecurityProviders=com.sun.security.sasl.Provider -J--add-exports=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED --exclude-config io\.netty\.netty-codec /META-INF/native-image/io\.netty/netty-codec/generated/handlers/reflect-config\.json --exclude-config io\.netty\.netty-handler /META-INF/native-image/io\.netty/netty-handler/generated/handlers/reflect-config\.json infinispan-client-micrometer-1.0.0-SNAPSHOT-runner -jar infinispan-client-micrometer-1.0.0-SNAPSHOT-runner.jar
========================================================================================================================
GraalVM Native Image: Generating 'infinispan-client-micrometer-1.0.0-SNAPSHOT-runner' (executable)...
========================================================================================================================
Warning: Could not register method io.quarkus.micrometer.runtime.export.PrometheusMeterRegistryProvider.configure(PrometheusRuntimeConfig) for reflection. Reason: java.lang.NoClassDefFoundError: io/micrometer/prometheus/PrometheusNamingConvention.
[1/7] Initializing...                                                                                    (9.7s @ 0.31GB)
 Version info: 'GraalVM 22.3.1 Java 17 CE'
 Java version info: '17.0.6+10-jvmci-22.3-b13'
 C compiler: cc (apple, x86_64, 14.0.3)
 Garbage collector: Serial GC
 4 user-specific feature(s)
 - io.quarkus.caffeine.runtime.graal.CacheConstructorsFeature
 - io.quarkus.infinispan.client.runtime.graal.DisableLoggingFeature: Disables INFO logging during the analysis phase
 - io.quarkus.runner.Feature: Auto-generated class by Quarkus from the existing extensions
 - io.quarkus.runtime.graal.DisableLoggingFeature: Disables INFO logging during the analysis phase
[2/7] Performing analysis...  []                                                                        (10.4s @ 1.47GB)
   4,380 (83.25%) of  5,261 classes reachable
   5,491 (50.47%) of 10,880 fields reachable
  18,061 (76.23%) of 23,692 methods reachable
     181 classes,     0 fields, and     0 methods registered for reflection

Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.handle(io.vertx.ext.web.RoutingContext)
Parsing context:
   at io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.handle(PrometheusHandler.java:27)
   at io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.handle(PrometheusHandler.java:18)
   at io.vertx.ext.web.impl.RouteState.handleFailure(RouteState.java:1290)
   at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:174)
   at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
   at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:68)
   at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
   at io.vertx.mutiny.ext.web.Router.handle(Router.java:75)
   at io.vertx.mutiny.ext.web.Router.accept(Router.java:486)
   at io.vertx.mutiny.ext.web.Router.accept(Router.java:27)
   at java.util.Collections$2.tryAdvance(Collections.java:4853)
   at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
   at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
   at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
   at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
   at java.util.stream.IntPipeline.findFirst(IntPipeline.java:552)
   at java.text.DecimalFormatSymbols.findNonFormatChar(DecimalFormatSymbols.java:844)
   at java.text.DecimalFormatSymbols.initialize(DecimalFormatSymbols.java:823)
   at java.text.DecimalFormatSymbols.<init>(DecimalFormatSymbols.java:115)
   at sun.util.locale.provider.DecimalFormatSymbolsProviderImpl.getInstance(DecimalFormatSymbolsProviderImpl.java:85)
   at java.text.DecimalFormatSymbols.getInstance(DecimalFormatSymbols.java:185)
   at java.util.Formatter.zero(Formatter.java:2450)
   at java.util.Formatter$FormatSpecifier.getZero(Formatter.java:4450)
   at java.util.Formatter$FormatSpecifier.localizedMagnitude(Formatter.java:4466)
   at java.util.Formatter$FormatSpecifier.localizedMagnitude(Formatter.java:4455)
   at java.util.Formatter$FormatSpecifier.print(Formatter.java:4034)
   at java.util.Formatter$FormatSpecifier.print(Formatter.java:3984)
   at java.util.Formatter$FormatSpecifier.printDateTime(Formatter.java:3009)
   at java.util.Formatter$FormatSpecifier.print(Formatter.java:2911)
   at java.util.Formatter.format(Formatter.java:2678)
   at java.util.Formatter.format(Formatter.java:2625)
   at java.lang.String.format(String.java:4141)
   at jdk.internal.util.Preconditions.outOfBoundsMessage(Preconditions.java:188)
   at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
   at jdk.internal.util.Preconditions.outOfBoundsCheckFromIndexSize(Preconditions.java:82)
   at jdk.internal.util.Preconditions.checkFromIndexSize(Preconditions.java:361)
   at java.util.Objects.checkFromIndexSize(Objects.java:411)

	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:153)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:104)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:83)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraph(MethodTypeFlow.java:65)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultVirtualInvokeTypeFlow.java:109)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:562)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis$1.run(PointsToAnalysis.java:488)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool.externalHelpQuiescePool(ForkJoinPool.java:2104)
	at java.base/java.util.concurrent.ForkJoinPool.awaitQuiescence(ForkJoinPool.java:3321)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.complete(CompletionExecutor.java:243)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.doTypeflow(PointsToAnalysis.java:541)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:529)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:143)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:745)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:578)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:580)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: io.micrometer.prometheus.PrometheusMeterRegistry. This error is reported at image build time because class io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler is registered for linking at image build time by command line
	at parsing io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.setup(PrometheusHandler.java:67)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2518)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:110)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3393)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3345)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3190)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1138)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1030)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:97)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:84)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:446)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:135)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:685)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.lookupEncodedGraph(InlineBeforeAnalysis.java:180)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.doInline(PEGraphDecoder.java:1162)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.tryInline(PEGraphDecoder.java:1145)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.trySimplifyInvoke(PEGraphDecoder.java:1003)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.handleInvoke(PEGraphDecoder.java:957)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:817)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysis.java:240)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:548)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:833)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:98)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:179)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:349)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:93)
	... 21 more
Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: io.micrometer.prometheus.PrometheusMeterRegistry. This error is reported at image build time because class io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler is registered for linking at image build time by command line
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.reportUnresolvedElement(SharedGraphBuilderPhase.java:333)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedType(SharedGraphBuilderPhase.java:288)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedLoadConstant(SharedGraphBuilderPhase.java:259)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genLoadConstant(BytecodeParser.java:3945)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5124)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3385)
	... 46 more
------------------------------------------------------------------------------------------------------------------------
                        0.6s (2.7% of total time) in 18 GCs | Peak RSS: 1.99GB | CPU load: 5.95
========================================================================================================================
Failed generating 'infinispan-client-micrometer-1.0.0-SNAPSHOT-runner' after 20.3s.
```

`quarkus-micrometer-registry-prometheus` is needed to compile succesfully into native when using combination of infinispan-client + micrometer. But when using just infinispan-client or just micrometer no additional deps are needed.

### Expected behavior

infinispan-client + micrometer combination compiles in native

### Actual behavior

infinispan-client + micrometer combination doesn't compile in native

### How to Reproduce?

 - mvn -Dquarkus-plugin.version=3.0.3.Final -Dquarkus.platform.artifact-id=quarkus-bom io.quarkus:quarkus-maven-plugin:3.0.3.Final:create -DprojectGroupId=io.quarkus.qe -DprojectArtifactId=infinispan-client-micrometer -DprojectVersion=1.0.0-SNAPSHOT -DplatformGroupId=io.quarkus -DplatformArtifactId=quarkus-bom -Dextensions=infinispan-client,micrometer -DquarkusRegistryClient=false
 - cd infinispan-client-micrometer
 - mvn clean package -Dnative


### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

22.3

### Quarkus version or git rev

3.0.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/PrometheusRegistryProcessor.java
+++ b/extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/PrometheusRegistryProcessor.java
@@ -39,7 +39,8 @@ public class PrometheusRegistryProcessor {
         MicrometerConfig mConfig;
 
         public boolean getAsBoolean() {
-            return REGISTRY_CLASS != null && mConfig.checkRegistryEnabledWithDefault(mConfig.export.prometheus);
+            return (REGISTRY_CLASS != null) && QuarkusClassLoader.isClassPresentAtRuntime(REGISTRY_CLASS_NAME)
+                    && mConfig.checkRegistryEnabledWithDefault(mConfig.export.prometheus);
         }
     }
 

